File: app/src/main/java/com/sivakar/eyerefresh/models/Event.java
package com.sivakar.eyerefresh.models;

import java.io.Serializable;

public enum Event implements Serializable {
    SCHEDULING_TURNED_ON,
    SCHEDULING_PAUSED,
    REMINDER_DUE,
    SNOOZE_REQUESTED,
    REFRESH_STARTED,
    REFRESH_TIME_UP,
    REFRESH_COMPLETED,
    REFRESH_CANCELLED,
    OPEN_APP
}

File: app/src/main/java/com/sivakar/eyerefresh/core/Event.kt
package com.sivakar.eyerefresh.core

import java.io.Serializable

sealed class AppEvent : Serializable {
    data object SchedulingTurnedOn : AppEvent()
    data object SchedulingPaused : AppEvent()
    data object RefreshDue : AppEvent()
    data object SnoozeRequested : AppEvent()
    data object RefreshStarted : AppEvent()
    data object RefreshTimeUp : AppEvent()
    data object MarkRefreshCompleted : AppEvent()
    data object RefreshCouldNotHappen : AppEvent()
    
    companion object {
        fun fromString(eventString: String): AppEvent? {
            return when (eventString) {
                "SchedulingTurnedOn" -> SchedulingTurnedOn
                "SchedulingPaused" -> SchedulingPaused
                "RefreshDue" -> RefreshDue
                "SnoozeRequested" -> SnoozeRequested
                "RefreshStarted" -> RefreshStarted
                "RefreshTimeUp" -> RefreshTimeUp
                "MarkRefreshCompleted" -> MarkRefreshCompleted
                "RefreshCouldNotHappen" -> RefreshCouldNotHappen
                else -> null
            }
        }
    }
}